src/web.rs

```diff
@@ -37,27 +61,63 @@ impl DataHandler {
     }
 }
 
+#[cfg(feature = "gzip")]
+macro_rules! return_gzip_or_not {
+    ($req:expr, $content:expr, $typ:expr) => {{
+        let mut use_gzip = false;
+
+        if let Some(raw_accept_encoding) = $req.headers.get_raw("accept-encoding") {
+            for accept_encoding in raw_accept_encoding {
+                match ::std::str::from_utf8(accept_encoding).map(|s| s.to_lowercase()) {
+                    Ok(ref s) if s.contains("gzip") => {
+                        use_gzip = true;
+                        break;
+                    }
+                    _ => continue,
+                }
+            }
+        }
+        if !use_gzip {
+            Ok(Response::with((status::Ok, $typ.parse::<Mime>().unwrap(), $content)))
+        } else {
+            let mut res = Response::new();
+            res.status = Some(status::Ok);
+            res.body = Some(Box::new(GzipContent(Box::new($content))));
+            res.headers.append_raw("content-type", $typ.as_bytes().to_vec());
+            res.headers.append_raw("content-encoding", vec![b'g', b'z', b'i', b'p']);
```

Comment:
@GuillaumeGomez it's not a big deal but we can use: [ContentType](https://docs.rs/iron/0.6.0/iron/headers/struct.ContentType.html) and [ContentEncoding](https://docs.rs/iron/0.6.0/iron/headers/struct.ContentEncoding.html) types here. I can make this change myself if you want.

You can see some example usage in [here](https://github.com/onur/docs.rs/blob/master/src/web/file.rs#L51-L53).